lib/elm-test.js

```diff
@@ -205,32 +205,29 @@ function runTests(generatedCodeDir /*: string */, testFile /*: string */) {
       ? '\\\\.\\pipe\\elm_test-' + process.pid + '-' + runsExecuted
       : '/tmp/elm_test-' + process.pid + '.sock';
 
-  return Compile.compile(
+  await Compile.compile(
     testFile,
     dest,
     args.verbose,
     pathToElmBinary,
     args.report
   )
-    .then(function () {
-      return Generate.prepareCompiledJsFile(pipeFilename, dest).then(
-        function () {
-          return Supervisor.run(
-            packageInfo.version,
-            pipeFilename,
-            report,
-            processes,
-            dest,
-            args.watch,
-            Compile.isMachineReadableReporter(report)
-          );
-        }
-      );
-    })
-    .catch(function (error) {
+  await Generate.prepareCompiledJsFile(pipeFilename, dest)
+  try {
```

Comment:
```suggestion
  try {
    await Generate.prepareCompiledJsFile(pipeFilename, dest)
```

The `.catch` in the promise chain catches errors from _all_ previous promises not just the last one.